open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java

```diff
@@ -248,20 +201,27 @@ protected void upsertExternalRelationship(String userId, String firstGUID, Strin
 
         String externalSourceGUID = dataEngineRegistrationHandler.getExternalDataEngine(userId, externalSourceName);
 
-        Optional<Relationship> relationship = findRelationship(userId, firstGUID, secondGUID, firstEntityTypeName, relationshipTypeName);
+        Optional<Relationship> relationship = findRelationship(userId, firstGUID, secondGUID, firstEntityTypeName,
+                secondEntityTypeName, relationshipTypeName);
         if (relationship.isEmpty()) {
 
             TypeDef relationshipTypeDef = repositoryHelper.getTypeDefByName(userId, relationshipTypeName);
-            repositoryHandler.createExternalRelationship(userId, relationshipTypeDef.getGUID(), externalSourceGUID, externalSourceName,
-                    firstGUID, secondGUID, relationshipProperties, methodName);
+
+            String relationshipGUID = genericHandler.linkElementToElement(userId, externalSourceGUID, externalSourceName, firstGUID,
```

Comment:
relationshipGUID is never used also. Not assigning the returned value of linkElementToElement should fix it